App.tsx

```diff
@@ -1,14 +1,23 @@
-// App.tsx or index.js
 import React from 'react';
-import { SafeAreaView } from 'react-native';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
 import PhotoGalleryScreen from './PhotoGalleryScreen';
+import ImageDetailsScreen from './ImageDetailsScreen';
+import ImageModalScreen from './ImageModalScreen'; // Import the ImageModalScreen
+import { RootStackParamList } from './types';
+
+const Stack = createStackNavigator<RootStackParamList>();
 
 const App = () => {
   return (
-    <SafeAreaView style={{ flex: 1 }}>
-      <PhotoGalleryScreen />
-    </SafeAreaView>
+    <NavigationContainer>
+      <Stack.Navigator initialRouteName="PhotoGallery">
+        <Stack.Screen name="PhotoGallery" component={PhotoGalleryScreen} />
+        <Stack.Screen name="ImageDetail" component={ImageDetailsScreen} />
+        <Stack.Screen name="ImageModal" component={ImageModalScreen} />
```

Comment:
This component should be presented as a modal using React Navigation 

`<Stack.Screen name="ImageModal" component={ImageModalScreen} options={{ presentation: 'modal' }}
 />`

You can take a look at "Presenting a screen modally" https://neit.instructure.com/courses/57491/pages/live-coding-react-navigation-+-stack-navigators?module_item_id=2592249